src/libpriv/rpmostree-core.c

```diff
@@ -2455,12 +2452,49 @@ rpmostree_context_commit_tmprootfs (RpmOstreeContext      *self,
         g_variant_builder_add (&metadata_builder, "{sv}", "rpmostree.clientlayer",
                                g_variant_new_boolean (TRUE));
 
+        if (!self->empty)
+          {
+            g_autoptr(GVariant) pkgs =
+              g_variant_dict_lookup_value (self->spec->dict, "packages",
+                                           G_VARIANT_TYPE ("as"));
+            g_assert (pkgs);
+            g_variant_builder_add (&metadata_builder, "{sv}",
+                                   "rpmostree.packages", pkgs);
+          }
+        else
+          {
+            const char *const* p = { NULL };
+            g_variant_builder_add (&metadata_builder, "{sv}",
+                                   "rpmostree.packages",
+                                   g_variant_new_strv (p, -1));
+          }
+
         /* be nice to our future selves */
         g_variant_builder_add (&metadata_builder, "{sv}",
                                "rpmostree.clientlayer_version",
-                               g_variant_new_uint32 (0));
+                               g_variant_new_uint32 (1));
+      }
+    else if (assemble_type == RPMOSTREE_ASSEMBLE_TYPE_SERVER_BASE)
+      {
+        g_autoptr(GVariant) spec_v =
+          g_variant_ref_sink (rpmostree_treespec_to_variant (self->spec));
+
+        g_variant_builder_add (&metadata_builder, "{sv}", "rpmostree.spec",
+                              spec_v);
+
+        g_variant_builder_add (&metadata_builder, "{sv}", "rpmostree.serverbase",
+                               g_variant_new_boolean (TRUE));
+
+        g_variant_builder_add (&metadata_builder, "{sv}",
+                               "rpmostree.serverbase_version",
```

Comment:
Kind of wonder if we need both the flag and the version...we could just combine them so `rpmostree.serverbase` is of type `u`.  Admittedly not symmetric with clientlayer then.